app/Http/Controllers/CompetitionsController.php

```diff
@@ -39,7 +48,9 @@ public function show(Competition $competition)
     {
         $contest = Contest::find($competition->contest_id);
 
-        return view('competitions.show', compact('competition', 'contest'));
+        $users = $this->repository->getAll($competition->users->pluck('id')->toArray());
```

Comment:
The Cache retrieval requires an actual array of ids and you can't pass a collection. I would get angry when you would change the collection to an array when passing it to the view, haha :flushed: